backend/src/main/java/com/rota/api/StaffService.java

```diff
@@ -32,12 +69,26 @@
    * @return A list of member's engagements
    */
   public List<EngagementDto> getStaffEngagementsBetween(int staffId, Instant start, Instant end) {
-    Instant startTime = start == null ? Instant.MIN : start;
-    Instant endTime = end == null ? Instant.MAX : end;
     return engagementRepository.findByStaffId(staffId).stream()
-        .filter(engagement ->
-            !engagement.getStart().isBefore(startTime)
-                && !engagement.getEnd().isAfter(endTime))
+        .filter(isEngagementBetween(start, end))
+        .map(truncateEngagement(start, end))
+        .map(EngagementDto::fromEngagement)
+        .collect(Collectors.toList());
+  }
+
+  /**
+   * Returns all engagements between start and end dates (inclusive).
+   * Specifically engagements that start after the start date and end before the end date.
```

Comment:
I think you changed this so that it gets engagements that overlap now as well?

Reply:
Yes, forgot to update the comment